src/java.naming/share/classes/javax/naming/spi/NamingManager.java

```diff
@@ -113,108 +108,14 @@ public class NamingManager {
      */
     public static synchronized void setObjectFactoryBuilder(
```

Comment:
Perhaps remove the `synchronized` from this method and the getter, now that the synchronization has moved to the `NamingManagerHelper` class? I don't see any other instance synchronization usages in this class, so removing the `synchronized` should be OK.

Reply:
Agreed - I think it is safe to remove `synchronized`. I doubt that there is a code that relies on `synchronized (javax.naming.spi.NamingManager.class)` to prevent other code setting the factory builder (also it can be set only once), therefore it should be ok to remove `synchronized`.